pkgs/path/test/io_test.dart

```diff
@@ -43,81 +43,78 @@ void main() {
-    try {
-      expect(path.absolute('foo/bar'), equals(path.join(dir, 'foo/bar')));
+    expect(path.absolute('foo/bar'), equals(path.join(dir, 'foo/bar')));
+    expect(
+      path.absolute('foo/bar'),
+      equals(path.context.join(dir, 'foo/bar')),
+    );
+
+    io.Directory.current = path.dirname(dir);
+    expect(
+      path.normalize(path.absolute('foo/bar')),
+      equals(path.normalize(path.join(dir, '../foo/bar'))),
+    );
+    expect(
+      path.normalize(path.absolute('foo/bar')),
+      equals(path.normalize(path.context.join(dir, '../foo/bar'))),
+    );
+  }));
+
+  // Regression test for #35. This tests against the *actual* working directory
+  // rather than just a custom context because we do some processing in
+  // [path.current] that has clobbered the root in the past.
+  test(
+    'absolute works on root working directory',
+    currentDirHelper(() {
+      io.sleep(const Duration(seconds: 2));
+      io.Directory.current = path.rootPrefix(path.current);
+      io.sleep(const Duration(seconds: 2));
+
       expect(
-        path.absolute('foo/bar'),
-        equals(path.context.join(dir, 'foo/bar')),
+        path.relative(path.absolute('foo/bar'), from: path.current),
+        path.relative(path.absolute('foo/bar')),
       );
 
-      io.Directory.current = path.dirname(dir);
       expect(
         path.normalize(path.absolute('foo/bar')),
-        equals(path.normalize(path.join(dir, '../foo/bar'))),
+        equals(path.normalize(path.join(path.current, '../foo/bar'))),
       );
+
       expect(
         path.normalize(path.absolute('foo/bar')),
-        equals(path.normalize(path.context.join(dir, '../foo/bar'))),
+        equals(path.normalize(path.join(path.current, '../foo/bar'))),
       );
-    } finally {
-      io.Directory.current = dir;
-    }
-  });
-
-  // Regression test for #35. This tests against the *actual* working directory
-  // rather than just a custom context because we do some processing in
-  // [path.current] that has clobbered the root in the past.
-  test(
-    'absolute works on root working directory',
-    () {
-      final dir = path.current;
-      try {
-        io.Directory.current = path.rootPrefix(path.current);
-
-        expect(
-          path.relative(path.absolute('foo/bar'), from: path.current),
-          path.relative(path.absolute('foo/bar')),
-        );
-
-        expect(
-          path.normalize(path.absolute('foo/bar')),
-          equals(path.normalize(path.join(path.current, '../foo/bar'))),
-        );
-
-        expect(
-          path.normalize(path.absolute('foo/bar')),
-          equals(path.normalize(path.join(path.current, '../foo/bar'))),
-        );
-      } finally {
-        io.Directory.current = dir;
-      }
-    },
+    }),
     //TODO(kevmoo): figure out why this is failing on windows and fix!
     skip: io.Platform.isWindows ? 'Untriaged failure on Windows' : null,
   );
 }
+
+dynamic Function() currentDirHelper(dynamic Function() body) {
```

Comment:
Can do `R Function() currentDirHelper<R>(R Function() body) {` to avoid the `dynamic`. (But if it's always used in a context with no static type constraint, it doesn't matter. Could use `Object?` instead of `dynamic` then.)

Or it could just return `void`, and then you use it inside the test, instead of around the test callback:
```dart
test('....', () {
  currentDirHelper(() {
    // Actual body.
  });
}
```

Reply:
Done.